cmd/network/packet-capture.go

```diff
@@ -0,0 +1,322 @@
+	}
+	return true, ds, nil
+}
+
+// createPacketCaptureDaemonSet creates the given daemonset resource
+func createPacketCaptureDaemonSet(o *packetCaptureOptions, ds *appsv1.DaemonSet) error {
+	if err := o.kubeCli.Create(context.TODO(), ds); err != nil {
+		return fmt.Errorf("failed to create daemonset %s/%s: %v", ds.Namespace, ds.Name, err)
+	}
+	return nil
+}
+
+// deletePacketCaptureDaemonSet creates the given daemonset resource
+func deletePacketCaptureDaemonSet(o *packetCaptureOptions, ds *appsv1.DaemonSet) error {
+	if err := o.kubeCli.Delete(context.TODO(), ds); err != nil {
+		return fmt.Errorf("failed to delete daemonset %s/%s: %v", ds.Namespace, ds.Name, err)
+	}
+	return nil
+}
+
+// desiredPacketCaptureDaemonSet returns the desired daemonset read in from manifests
+func desiredPacketCaptureDaemonSet(o *packetCaptureOptions, key types.NamespacedName) *appsv1.DaemonSet {
+	ds := &appsv1.DaemonSet{}
+	t := true
+	ls := &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"app": key.Name,
+		},
+	}
+	ds.Name = key.Name
+	ds.Namespace = key.Namespace
+
+	ds.Spec.Selector = ls
+	ds.Spec.Template.Spec.NodeSelector = map[string]string{
+		o.nodeLabelKey: o.nodeLabelValue,
+	}
+	ds.Spec.Template.Labels = ls.MatchLabels
+	ds.Spec.Template.Spec.Tolerations = []corev1.Toleration{
+		{
+			Effect:   "NoSchedule",
+			Key:      o.nodeLabelKey,
+			Operator: "Exists",
+		},
+	}
+	ds.Spec.Template.Spec.Volumes = []corev1.Volume{
+		{
+			Name: "capture-output",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+	ds.Spec.Template.Spec.HostNetwork = true
+	ds.Spec.Template.Spec.InitContainers = []corev1.Container{
+		{
+			Name:            "init-capture",
+			Image:           packetCaptureImage,
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			Command:         []string{"/bin/bash", "-c", "tcpdump -G " + strconv.Itoa(o.duration) + " -W 1 -w /tmp/capture-output/capture.pcap -i vxlan_sys_4789 -nn -s0; sync"},
+			SecurityContext: &corev1.SecurityContext{Privileged: &t},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "capture-output",
+					MountPath: "/tmp/capture-output",
+					ReadOnly:  false,
+				},
+			},
+		},
+	}
+	ds.Spec.Template.Spec.Containers = []corev1.Container{
+		{
+			Name:            "copy",
+			Image:           packetCaptureImage,
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			Command:         []string{"/bin/bash", "-c", "trap : TERM INT; sleep infinity & wait"},
+			SecurityContext: &corev1.SecurityContext{Privileged: &t},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "capture-output",
+					MountPath: "/tmp/capture-output",
+					ReadOnly:  false,
+				},
+			},
+		},
+	}
+
+	return ds
+}
+
+func copyFilesFromPod(o *packetCaptureOptions, pod *corev1.Pod) error {
+	os.MkdirAll(outputDir, 0755)
+	cmd := exec.Command("oc", "cp", pod.Namespace+"/"+pod.Name+":/tmp/capture-output/capture.pcap", outputDir+"/"+pod.Name+".pcap")
```

Comment:
Should we consider adding a timestamp to these pcap files instead of just the pod name?

Reply:
Good suggestion. I changed the filename format to `<node name>-<iso 8601>.pcap`, example: `ip-10-0-149-114.us-west-2.compute.internal-20201214T215510.pcap`